frontend/talentsearch/src/js/components/pool/PoolAdvertisementPage.tsx

```diff
@@ -126,11 +135,36 @@ const PoolAdvertisement = ({ poolAdvertisement }: PoolAdvertisementProps) => {
     poolAdvertisement.nonessentialSkills,
   );
 
+  const [, executeMutation] = useCreateApplicationMutation();
+  const handleCreateApplication = () =>
+    executeMutation({ userId, poolId: poolAdvertisement.id })
+      .then((result) => {
+        if (result.data?.createApplication) {
+          navigate(paths.reviewApplication(result.data.createApplication.id));
+          toast(
+            intl.formatMessage({
+              defaultMessage: "Application created",
+              id: "U/ji+A",
+              description: "Application created successfully",
+            }),
+          );
+          return result.data?.createApplication.id;
+        }
+        return result.data?.createApplication;
+      })
+      .catch((result) => {
+        toast(
+          intl.formatMessage({
+            defaultMessage: "Error application creation failed",
+            id: "tlAiJm",
+            description: "Application creation failed",
+          }),
+        );
```

Comment:
Similar to the success toast but for error styling.

```suggestion
        toast.error(
          intl.formatMessage({
            defaultMessage: "Error application creation failed",
            id: "tlAiJm",
            description: "Application creation failed",
          }),
        );
```